roborock/cloud_api.py

```
@@ -158,7 +159,8 @@ async def async_disconnect(self) -> None:
             if disconnected_future := self._sync_disconnect():
                 # There are no errors set on this future
                 await disconnected_future
-            await self.event_loop.run_in_executor(None, self._mqtt_client.loop_stop)
+            loop = asyncio.get_running_loop()
```

My only thought is if there is any chance that an event loop doesn't exist at these points. But even for non-ha users, it should as it's needed for the web api so it should be okay

---

Yeah my thinking is that if it has, to exist to be correct. If it doesn't exist then failing loudly is best.

We can label this as a breaking change? Not sure how semantic commit handles this.. "refactor"?

---

I don't think we need to mark it as a breaking change - partially as i am also not sure how to do that with semantic release haha

It should be fine as most users will create the web api first and will get an event loop there if it doesn't exist.